### ISSUE TITLE
Exception in thread "main" com.fasterxml.jackson.databind.JsonMappingException: Invalid type=INT

### ISSUE THREAD

**Issue by cozima0210** [msgpack/msgpack-java#526]
When I use this library with my scala project, below Exception occurs. It seems that it occurs when I try to deserialize a value of BigDecimal serialized by Messagepack.

```build.sbt
scalaVersion := "2.13.3"

libraryDependencies ++= Seq(
  "com.fasterxml.jackson.module" %% "jackson-module-scala"      % "2.11.1",
  "org.msgpack"                  % "jackson-dataformat-msgpack" % "0.8.20"
)
```

```Main.scala
import com.fasterxml.jackson.databind.{DeserializationFeature, ObjectMapper}
import com.fasterxml.jackson.module.scala.DefaultScalaModule
import com.fasterxml.jackson.module.scala.experimental.ScalaObjectMapper
import org.msgpack.jackson.dataformat.MessagePackFactory

object Main extends App {

  case class Hoge(d: BigDecimal)

  val obj = Hoge(BigDecimal(0))

  val mapper = new ObjectMapper(new MessagePackFactory) with ScalaObjectMapper
  mapper.registerModule(DefaultScalaModule)
  mapper.enable(DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS)

  val bytes = mapper.writeValueAsBytes(obj)

  val hoge = mapper.readValue[Hoge](bytes)
}
```

```exception.sh
Exception in thread "main" com.fasterxml.jackson.databind.JsonMappingException: Invalid type=INT (through reference chain: Main$Hoge["d"])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:397)
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:356)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.wrapAndThrow(BeanDeserializerBase.java:1726)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:537)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:419)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1322)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:331)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:164)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4482)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3529)
	at com.fasterxml.jackson.module.scala.ScalaObjectMapper.readValue(ScalaObjectMapper.scala:203)
	at com.fasterxml.jackson.module.scala.ScalaObjectMapper.readValue$(ScalaObjectMapper.scala:202)
	at Main$$anon$1.readValue(Main.scala:13)
	at Main$.delayedEndpoint$Main$1(Main.scala:21)
	at Main$delayedInit$body.apply(Main.scala:7)
	at scala.Function0.apply$mcV$sp(Function0.scala:39)
	at scala.Function0.apply$mcV$sp$(Function0.scala:39)
	at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:17)
	at scala.App.$anonfun$main$1(App.scala:73)
	at scala.App.$anonfun$main$1$adapted(App.scala:73)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:553)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:551)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:920)
	at scala.App.main(App.scala:73)
	at scala.App.main$(App.scala:71)
	at Main$.main(Main.scala:7)
	at Main.main(Main.scala)
Caused by: java.lang.IllegalStateException: Invalid type=INT
	at org.msgpack.jackson.dataformat.MessagePackParser.getText(MessagePackParser.java:387)
	at com.fasterxml.jackson.module.scala.deser.BigNumberDeserializer.deserialize(ScalaNumberDeserializersModule.scala:20)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:542)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeWithErrorWrapping(BeanDeserializer.java:535)
	... 23 more
```
I have already found that the workaround is to fix `getText` in `MessagePackParser.java(msgpack/msgpack-java)` like below.
https://github.com/cozima0210/msgpack-java/commit/d11ecc5584440ab97d3520062de83c72b8167b2b

But I know that there is another workaround to solve it. It is to fix `deserialize` in `ScalaNumberDeserializersModule.scala(FasterXML/jackson-module-scala)` like below.
https://github.com/cozima0210/jackson-module-scala/commit/a6f7420bdddff1fb7a4e67ef13897e4b1d528c8c

Therefore, I want to hear you which repositories I should send PR. Thanks.

**Comment by cozima0210** [comment]
I also send a same [issue](https://github.com/FasterXML/jackson-module-scala/issues/458) to https://github.com/FasterXML/jackson-module-scala/.

**Comment by komamitsu** [comment]
I tried to reproduce this issue in Java, but I couldn't. Is there any way to reproduce this issue without jackson-module-scala?

**Comment by cozima0210** [comment]
@komamitsu Thanks for your reply. No, it isn't. If you tried to reproduce it with `java.math.BigDecimal`, please confirm this issue with `scala.math.BigDecimal`. Currently I don't understand which repository should be fixed to resolve this error.

**Comment by pjfanning** [comment]
@komamitsu the fix in https://github.com/cozima0210/msgpack-java/commit/d11ecc5584440ab97d3520062de83c72b8167b2b is to allow getText to work for numbers. It is not not uncommon for users to encode numbers as JSON strings because many JSON libs assume JSON numbers are doubles and double are only an approximation of the actual number - while in text you can provide the number with arbitrary precision.

**Comment by ktholen** [comment]
@komamitsu @pjfanning would it be possible to cut a new release? We just ran into this problem today and would like to use the new version in our application.

**Comment by komamitsu** [comment]
@ktholen Released 0.8.21 that includes this fix.

**Comment by ktholen** [comment]
Thank you! Works as expected!
